internal/stoker/stokercr/scanners/trivy.go

```diff
@@ -0,0 +1,154 @@
+package scanners
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/aquasecurity/trivy/pkg/commands/artifact"
+	"github.com/aquasecurity/trivy/pkg/commands/operation"
+	"github.com/aquasecurity/trivy/pkg/db"
+	fanaltypes "github.com/aquasecurity/trivy/pkg/fanal/types"
+	"github.com/aquasecurity/trivy/pkg/flag"
+	"github.com/aquasecurity/trivy/pkg/types"
+	"github.com/frantjc/sindri/internal/cache"
+	"github.com/frantjc/sindri/internal/stoker/stokercr"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/uuid"
+)
+
+type trivy struct {
+	runner   artifact.Runner
+	dbRepos  []name.Reference
+	cacheDir string
+}
+
+type TrivyOptions struct {
+	DBRepositories []string
+}
+
+type TrivyOption func(*TrivyOptions)
+
+func WithDBRepositories(repos []string) TrivyOption {
+	return func(o *TrivyOptions) {
+		o.DBRepositories = repos
+	}
+}
+
+func NewTrivy(ctx context.Context, opts ...TrivyOption) (*trivy, error) {
+	options := &TrivyOptions{}
+
+	for _, opt := range opts {
+		opt(options)
+	}
+
+	if len(options.DBRepositories) == 0 {
+		options.DBRepositories = []string{
+			"ghcr.io/aquasecurity/trivy-db:2",
+			"mirror.gcr.io/aquasec/trivy-db:2",
+		}
+	}
+
+	dbRepos := make([]name.Reference, 0, len(options.DBRepositories))
+	for _, repo := range options.DBRepositories {
+		ref, err := name.ParseReference(repo)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse repository %s: %w", repo, err)
+		}
+		dbRepos = append(dbRepos, ref)
+	}
+
+	cacheDir := fmt.Sprintf("%s/trivy", cache.Dir)
+
+	if err := operation.DownloadDB(
+		ctx,
+		"dev",
+		cacheDir,
+		dbRepos,
+		false,
+		false,
+		fanaltypes.RegistryOptions{},
+	); err != nil {
+		return nil, fmt.Errorf("failed to download Trivy DB: %w", err)
+	}
+
+	if err := db.Init(db.Dir(cacheDir)); err != nil {
+		return nil, fmt.Errorf("failed to initialize Trivy DB: %w", err)
+	}
+
+	runner, err := artifact.NewRunner(
+		ctx,
+		flag.Options{},
+		artifact.TargetContainerImage,
+	)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create Trivy runner: %w", err)
+	}
+
+	return &trivy{runner, dbRepos, cacheDir}, nil
+}
+
+func (s trivy) Scan(ctx context.Context, b bytes.Buffer) ([]stokercr.Vuln, error) {
+	d := fmt.Sprintf("%s/sindri", s.cacheDir)
+	p := fmt.Sprintf("%s/image-%s.tar", d, uuid.New())
```

Comment:
I'd prefer not to write this to a file if possible, instead passing the `io.Reader` directly